src/diffusers/pipelines/wuerstchen/pipeline_wuerstchen.py

```diff
@@ -0,0 +1,316 @@
+            seq_len = uncond_text_encoder_hidden_states.shape[1]
+            uncond_text_encoder_hidden_states = uncond_text_encoder_hidden_states.repeat(1, num_images_per_prompt, 1)
+            uncond_text_encoder_hidden_states = uncond_text_encoder_hidden_states.view(
+                batch_size * num_images_per_prompt, seq_len, -1
+            )
+            # done duplicates
+
+            # For classifier free guidance, we need to do two forward passes.
+            # Here we concatenate the unconditional and text embeddings into a single batch
+            # to avoid doing two forward passes
+            text_encoder_hidden_states = torch.cat([text_encoder_hidden_states, uncond_text_encoder_hidden_states])
+
+        return text_encoder_hidden_states
+
+    def check_inputs(
+        self, predicted_image_embeddings, text_encoder_hidden_states, do_classifier_free_guidance, device
+    ):
+        if not isinstance(text_encoder_hidden_states, torch.Tensor):
+            raise TypeError(
+                f"'text_encoder_hidden_states' must be of type 'torch.Tensor', but got {type(predicted_image_embeddings)}."
+            )
+        if isinstance(predicted_image_embeddings, np.ndarray):
+            predicted_image_embeddings = torch.Tensor(predicted_image_embeddings, device=device).to(
+                dtype=text_encoder_hidden_states.dtype
+            )
+        if not isinstance(predicted_image_embeddings, torch.Tensor):
+            raise TypeError(
+                f"'predicted_image_embeddings' must be of type 'torch.Tensor' or 'np.array', but got {type(predicted_image_embeddings)}."
+            )
+
+        if do_classifier_free_guidance:
+            assert (
+                predicted_image_embeddings.size(0) == text_encoder_hidden_states.size(0) // 2
+            ), f"'text_encoder_hidden_states' must be double the size of 'predicted_image_embeddings' in the first dimension, but {predicted_image_embeddings.size(0)} != {text_encoder_hidden_states.size(0)}."
+        else:
+            if predicted_image_embeddings.size(0) * 2 == text_encoder_hidden_states.size(0):
+                text_encoder_hidden_states = text_encoder_hidden_states.chunk(2)[0]
+            assert predicted_image_embeddings.size(0) == text_encoder_hidden_states.size(
+                0
+            ), f"'text_encoder_hidden_states' must be the size of 'predicted_image_embeddings' in the first dimension, but {predicted_image_embeddings.size(0)} != {text_encoder_hidden_states.size(0)}."
+
+        return predicted_image_embeddings, text_encoder_hidden_states
+
+    @torch.no_grad()
+    def encode_image(self, image):
+        return self.efficient_net(image)
+
+    @torch.no_grad()
+    def __call__(
+        self,
+        image_embeds: torch.Tensor,
+        prompt: Union[str, List[str]] = None,
+        negative_prompt: Optional[Union[str, List[str]]] = None,
+        num_inference_steps: dict[float, int] = {0.0: 12},
```

Comment:
`num_inference_steps` should not be a dict, it should always be of type `int` - let's try to change this. We need to ensure consistency across pipelines here